sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -421,68 +427,77 @@ export class MessageSender extends LinkEntity {
   }
 
   /**
-   * Initializes the sender session on the connection.
+   * Initializes the `ClientEntityContext.sender` associated with this
```

Comment:
We only set the context.sender in `MessageSender.create()` right?
As far as `open()` is concerned, it is only opening the link associated with `MessageSender`.
The `open()` is not aware of the fact that the `MessageSender` is also cached in the context.

I would recommend

```suggestion
   * Initializes the underlying AMQP sender link from rhea associated with this
```